src/components/access-type.spec.js

```diff
@@ -0,0 +1,388 @@
+                public: true,
+                created_at: "2021-06-17T10:14:49.926Z",
+                updated_at: "2021-06-17T10:14:49.926Z",
+                preferred_identity_provider: "email",
+                currency: "JPY",
+                target_amount: 30000,
+                target_reached: false,
+                collected_amount_cents: 0,
+                collected_amount: 0,
+                collected_amount_percentage: 0,
+                subscription_plans: [
+                  {
+                    id: 47760,
+                    subscription_group_id: 20760,
+                    duration_length: 5,
+                    price_cents: 10000,
+                    price_currency: "JPY",
+                    created_at: "2021-06-17T10:16:02.149Z",
+                    updated_at: "2021-06-17T10:16:02.149Z",
+                    duration_unit: "days",
+                    description: "yen currency",
+                    title: "Campaign Omise",
+                    recurring: false,
+                    enabled: true,
+                    trial_period_enabled: false,
+                    supported_payment_providers: ["omise"],
+                    assets: [
+                      {
+                        title: "full-site",
+                        type: "site",
+                      },
+                    ],
+                    display_assets: [
+                      {
+                        title: "full-site",
+                        type: "site",
+                      },
+                    ],
+                  },
+                ],
+              },
+            ]);
+          } else if (host.includes("/api/v1/access-token/integrations")) {
+            return Promise.resolve({
+              headers: {
+                "x-integration-token":
+                  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdmF0YXItdXJsIjpudWxsLCJuYW1lIjoiYWJjIiwiaWQiOjE2OTQ1MzEsImVtYWlsIjoiM2QyOTM5MWUtNDAzMy00NzhmLTlhOGMtYTc4ZjVmMTM0ODdmQGVtYWlsLndlYmhvb2suc2l0ZSIsImZpcnN0LW5hbWUiOm51bGwsImxhc3QtbmFtZSI6bnVsbCwidGltZXN0YW1wIjoxNjIzOTI4NzIxNzYzLCJpYXQiOjE2MjM5Mjg3MjF9.gJjwrDDaaNBFQNfrDcdzczCIsN_pxVD1njYXA-6XxOs",
+              },
+            });
+          }
+        },
+      })
+    );
+  });
+
+  let selectedPlan = {
+    id: 47742,
+    duration_length: 5,
+    price_cents: 3000,
+    price_currency: "THB",
+    created_at: "2021-06-15T08:31:56.348Z",
+    updated_at: "2021-06-16T11:58:23.996Z",
+    duration_unit: "days",
+    description: "Selected Plan",
+    title: "Selected Plan",
+    recurring: true,
+    enabled: true,
+    trial_period_enabled: false,
+    supported_payment_providers: ["omise"],
+  };
+  let instance;
+
+  it("Successful omise recurring payment", () => {
+    const container = render(
+      <Provider
+        store={createStore((x) => x, {
+          paymentOptions: {
+            omise: {
+              action: "pay",
+              proceed: async (i) => {
+                return { proceed: (j) => j };
```

Comment:
Please change to more descriptive variable names instead of `x`, `i`, and `j` in all instances